pinot-spi/src/main/java/org/apache/pinot/spi/utils/CommonConstants.java

```diff
@@ -218,11 +225,16 @@ public static class Broker {
     public static final String CONFIG_OF_BROKER_GROUPBY_TRIM_THRESHOLD = "pinot.broker.groupby.trim.threshold";
     public static final int DEFAULT_BROKER_GROUPBY_TRIM_THRESHOLD = 1_000_000;
 
+    // Configure the request handler type used by broker to handler inbound query request.
+    // NOTE: the request handler type refers to the communication between Broker and Server.
+    public static final String BROKER_REQUEST_HANDLER_TYPE_JSON_OVERRIDE_KEY = "handler.type";
     public static final String BROKER_REQUEST_HANDLER_TYPE = "pinot.broker.request.handler.type";
     public static final String NETTY_BROKER_REQUEST_HANDLER_TYPE = "netty";
     public static final String GRPC_BROKER_REQUEST_HANDLER_TYPE = "grpc";
+    public static final String MULTI_STAGE_BROKER_REQUEST_HANDLER_TYPE = "multi_stage";
```

Comment:
(nit) not sure if we use underscore otherwise in configs. may be just use `multistage` ?

Reply:
Thinking more about this, I don't think a request handler specific config is probably not needed for the multistage engine.

Our current goal is to have both the existing request handlers (GRPC and Netty based) to be able to leverage the new engine as needed. Correct ?

So, then we just check the feature flag (`CONFIG_OF_MULTI_STAGE_ENGINE_ENABLED`) config in `BaseBrokerStarter` (like already being done in the PR) and create the multistage engine request handler and put it under the delegate.

Now, the delegate uses the request specific key for the given query request to determine which request handler to be invoked. So, I don't see us using this config for now. Thoughts ?

Reply:
the reason is `multi-stage` is yet another handler type as we defined in previous `RequestHandler` concrete class. also we have the `OVERRIDE_KEY` configuration in JSON payload from the request itself. 

If we opted to use a boolean flag like `use.multistage.engine=true` in the JSON payload then we don't need this . I am open to either route.